wallet/index.ts

```diff
@@ -71,40 +70,46 @@ export async function newWallet(): Promise<BaseWallet> {
   return walletFromSeedPhrase({ mnemonic, index: 0n, network: 'Mainnet' });
 }
 
-export async function walletFromSeedPhrase({
-  mnemonic,
+const waitFrame = () => (requestAnimationFrame ? new Promise(requestAnimationFrame) : Promise.resolve());
+
+export async function getWalletFromRootNode({
   index,
   network,
+  rootNode,
+  master,
 }: {
-  mnemonic: string;
   index: bigint;
   network: NetworkType;
-}): Promise<BaseWallet> {
-  const seed = await bip39.mnemonicToSeed(mnemonic);
-  const rootNode = bip32.fromSeed(Buffer.from(seed));
-
+  rootNode: BIP32Interface;
+  master: BIP32Interface;
+}): Promise<Omit<BaseWallet, 'masterPubKey' | 'seedPhrase'>> {
   const deriveStxAddressKeychain = deriveStxAddressChain(
     network === 'Mainnet' ? ChainID.Mainnet : ChainID.Testnet,
-    index
+    index,
   );
 
-  const { address, privateKey } = deriveStxAddressKeychain(rootNode);
+  await waitFrame();
+  const { address, privateKey } = await deriveStxAddressKeychain(rootNode);
   const stxAddress = address;
 
-  const master = bip32.fromSeed(seed);
-  const masterPubKey = master.publicKey.toString('hex');
   const stxPublicKey = publicKeyToString(getPublicKey(createStacksPrivateKey(privateKey)));
 
   // derive segwit btc address
+  await waitFrame();
   const btcChild = master.derivePath(getBitcoinDerivationPath({ index, network }));
+
+  await waitFrame();
```

Comment:
i am not sure about this one did it make a notable difference in performance?

Reply:
in mobile the difference is noticeable. we gain several frames between tasks with this

Reply:
removed